pkg/common/cns-lib/vsphere/virtualcenter.go

```diff
@@ -315,6 +315,9 @@ func (vc *VirtualCenter) connect(ctx context.Context, requestNewSession bool) er
 func (vc *VirtualCenter) ListDatacenters(ctx context.Context) (
 	[]*Datacenter, error) {
 	log := logger.GetLogger(ctx)
+	if vc.Client == nil {
+		return nil, logger.LogNewError(log, "failed to list datacenters due to VC client is reset")
```

Comment:
VC client is reset => VC client is not set.

Reply:
any other places in the `pkg/common/cns-lib/vsphere/virtualcenter.go` we want to handle usage of nil vc.Client?

Reply:
>any other places in the pkg/common/cns-lib/vsphere/virtualcenter.go we want to handle usage of nil vc.Client?

That's the only place as as far as I can see.